### PR TITLE
Release 1.31.0

### PR DESCRIPTION
**Pull request by sfc-gh-dmatthews**
## 📚 Context
New content for release 1.31.0

## 🧠 Description of Changes
* Added `st.page_link`
* Added `st.write_stream`
* Updated `st.chat_input` and tutorial on building chat apps

## 💥 Impact
- [X] Not small

## 🌐 References
Notion task: [Docs for Release 1.31.0](https://www.notion.so/snowflake-corp/Docs-for-Release-1-31-0-35058e12abc142f697d3009538672072?pvs=4)

**Contribution License Agreement**
By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

### CODE REVIEW

**Review by sfc-gh-jcarroll — APPROVED** [review]
LGTM. Nice work!

File: content/library/api/widgets/page_link.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+title: st.page_link
+slug: /library/api-reference/widgets/st.page_link
+---
+
+<Autofunction function="streamlit.page_link" />
```
Comment (sfc-gh-jcarroll) [inline content/library/api/widgets/page_link.md:6]: Note for later follow-up: once we have the full nav guide, we might want to link it from here too?

File: content/library/changelog.md [hunk 2]
```diff
@@ -17,6 +17,32 @@ pip install --upgrade streamlit
 
 </Tip>
 
+## **Version 1.31.0**
+
+_Release date: February 1, 2024_
+
+**Highlights**
+
+- 🔗 Introducing `st.page_link`! Now you can build your own, custom navigation menus for your multipage apps. Check out [our docs](/library/api-reference/widgets/st.page_link) to see how.
```
Comment (sfc-gh-skekre) [inline content/library/changelog.md:26]: Rephrasing for clarity:

```suggestion
- 🔗 Introducing `st.page_link`! Now, you can build custom navigation menus for your multipage apps. Check out [our docs](/library/api-reference/widgets/st.page_link) to see how.
```

File: content/library/api/widgets/page_link.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+title: st.page_link
+slug: /library/api-reference/widgets/st.page_link
+---
```
Comment (sfc-gh-skekre) [inline content/library/api/widgets/page_link.md:4]:
```suggestion
---
title: st.page_link
slug: /library/api-reference/widgets/st.page_link
description: st.page_link displays a link to another page in a multipage app or to an external page.
---
```

**Review by snehankekre — APPROVED** [review]
:shipit:

File: content/library/api/widgets/page_link.md (hunk 1 above)
Comment (sfc-gh-dmatthews) [reply content/library/api/widgets/page_link.md:13]: Added.

**Pull request merged**